File: src/com/main/java/SafeMethodCalls.java
package com.main.java;

import com.main.kotlin.Customer;
import com.main.kotlin.Product;

import java.util.Optional;

public class SafeMethodCalls {

    public static void main(String[] args) {

        CustomerService customerService = new CustomerServiceJava();

        String passportNumber = "";
        Optional<Customer> customer = customerService.getCustomer(passportNumber);
        Optional<Product> lastPurchase = customer.map(c -> c.lastPurchase());

        Optional<Product> product1 = customerService.getCustomer(passportNumber)
                .map(Customer::lastPurchase);
    }
}

File: src/com/main/kotlin/SafeMethodCalls.kt
package com.main.kotlin


fun main() {
    val customerService = CustomerServiceKotlin()

    val customer: Customer? = customerService.getCustomer(passportNumber)
    val product: Product? = customer?.lastPurchase()

    val product1: Product? = customerService.getCustomer(passportNumber)?.lastPurchase()
    
}



